pkg/controllers/hypernode/hypernode_handler.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2025 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hypernode
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	topologyv1alpha1 "volcano.sh/apis/pkg/apis/topology/v1alpha1"
+)
+
+func (hn *hyperNodeController) addHyperNode(obj interface{}) {
+	hyperNode, ok := obj.(*topologyv1alpha1.HyperNode)
+	if !ok {
+		klog.ErrorS(nil, "Cannot convert to *topologyv1alpha1.HyperNode", "obj", obj)
+		return
+	}
+	klog.V(3).InfoS("Add HyperNode", "name", hyperNode.Name)
+	hn.enqueueHyperNode(hyperNode)
+}
+
+func (hn *hyperNodeController) updateHyperNode(oldObj, newObj interface{}) {
+	hyperNode, ok := newObj.(*topologyv1alpha1.HyperNode)
+	if !ok {
+		klog.ErrorS(nil, "Cannot convert to *topologyv1alpha1.HyperNode", "obj", newObj)
+		return
+	}
+	klog.V(3).InfoS("Update HyperNode", "name", hyperNode.Name)
+	hn.enqueueHyperNode(hyperNode)
+}
+
+func (hn *hyperNodeController) enqueueHyperNode(hyperNode *topologyv1alpha1.HyperNode) {
+	key, err := cache.MetaNamespaceKeyFunc(hyperNode)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get key for HyperNode", "hyperNode", hyperNode.Name)
+		return
+	}
+	hn.hyperNodeQueue.Add(key)
+}
+
+// processHyperNodeQueue processes items from the hyperNode work queue
+func (hn *hyperNodeController) processHyperNodeQueue() {
+	for {
+		key, shutdown := hn.hyperNodeQueue.Get()
+		if shutdown {
+			klog.InfoS("HyperNode queue has been shut down")
+			return
+		}
+
+		func() {
+			defer hn.hyperNodeQueue.Done(key)
+			err := hn.syncHyperNodeStatus(key)
+			if err != nil {
+				klog.ErrorS(err, "Error syncing HyperNode", "key", key)
+				hn.hyperNodeQueue.AddRateLimited(key)
```

Comment:
Do we need to judge the times of requeue? If the times of requeue exceed the maxRetries times, then we need to forget the key

Reply:
I think we should retry, as we should update this field correctly, and the retry is a backoff retry, we there will be no too many API retry happen.